graylog2-server/src/test/java/org/graylog2/plugin/utilities/date/DateTimeConverterTest.java

```diff
@@ -0,0 +1,158 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class DateTimeConverterTest {
+
+    @Test
+    void convertFromDateTime() {
+        final DateTime input = new DateTime(2021, 8, 19, 12, 0, DateTimeZone.UTC);
+
+        final DateTime output = DateTimeConverter.convertToDateTime(input);
+
+        assertThat(output).isEqualTo(input);
+    }
+
+    @Test
+    void convertFromDate() {
+        final long currentTimeMillis = System.currentTimeMillis();
+        final Date input = new Date(currentTimeMillis);
+
+        final DateTime output = DateTimeConverter.convertToDateTime(input);
+
+        final DateTime expectedOutput = new DateTime(currentTimeMillis, DateTimeZone.UTC);
+        assertThat(output).isEqualTo(expectedOutput);
+    }
+
+    @Test
+    void convertFromZonedDateTimeNewYork() {
+        final ZoneId newYork = ZoneId.of("America/New_York");
+        final ZonedDateTime input = ZonedDateTime.of(2020, 10, 24, 10, 59, 0, 0, newYork);
+
+        final DateTime output = DateTimeConverter.convertToDateTime(input);
+
+        final DateTime expectedOutput = new DateTime(2020, 10, 24, 10, 59, DateTimeZone.forID("America/New_York"));
+        assertThat(output).isEqualTo(expectedOutput);
+    }
+
+    @Test
+    void convertFromZonedDateTimeUTC() {
+        final ZonedDateTime input = ZonedDateTime.of(2020, 10, 24, 10, 59, 0, 0, ZoneOffset.UTC);
+
+        final DateTime output = DateTimeConverter.convertToDateTime(input);
+
+        final DateTime expectedOutput = new DateTime(2020, 10, 24, 10, 59, DateTimeZone.UTC);
+        assertThat(output).isEqualTo(expectedOutput);
+    }
+
+    @Test
+    void convertFromOffsetDateTime() {
+        final OffsetDateTime input = OffsetDateTime.of(2021, 11, 20, 14, 50, 10, 0, ZoneOffset.UTC);
+
+        final DateTime output = DateTimeConverter.convertToDateTime(input);
+
+        final DateTime expectedOutput = new DateTime(2021, 11, 20, 14, 50, 10, DateTimeZone.UTC);
+        assertThat(output).isEqualTo(expectedOutput);
+    }
+
+    @Test
+    void convertFromOffsetDateTimeWithNonUTCOffset() {
+        final OffsetDateTime input = OffsetDateTime.of(2021, 11, 20, 14, 50, 10, 0, ZoneOffset.of("+1"));
+
+        final DateTime output = DateTimeConverter.convertToDateTime(input);
+
+        final DateTime expectedOutput = new DateTime(2021, 11, 20, 13, 50, 10, DateTimeZone.UTC);
+        assertThat(output).isEqualTo(expectedOutput);
+    }
+
+    @Test
+    @SuppressForbidden("Comparing twice with default timezone is okay in tests")
+    void convertFromLocalDateTime() {
+        final LocalDateTime input = LocalDateTime.of(2021, Month.AUGUST, 19, 12, 0);
+
+        final DateTime output = DateTimeConverter.convertToDateTime(input);
+
+        // represented in local timezone, but then converted to UTC
+        //final DateTime expectedOutput = new DateTime(2021, 8, 19, 12, 0).withZone(DateTimeZone.UTC);
```

Comment:
Leftover comment but it highlights what we need to decide: should we always return UTC or do we keep the time zone (in this specific case the system default zone I assume)

Reply:
Thanks, I was opting for keeping the timezone. So I'll remove the wrong comment